File: CPP-Source/Day9.cpp
//DAY 9 OF HACKERRANK 30 DAY CHALLENGE: RECURSION (C++)
//NOTE: Commented out included header from HackerRank for the sake of compilation.

//#include <bits/stdc++.h>
#include <iostream>
using namespace std;

int factorial(int n) {
	int result = 0;
	if (n == 1) {
		result = 1;
	}
	else {
		result = factorial(n - 1) * n;
	}
	return result;
}

int main() {
	int n;
	cin >> n;
	int result = factorial(n);
	cout << result << endl;
	return 0;
}

File: Java-Source/Day9.java
//DAY 9 OF HACKERRANK 30 DAY CHALLENGE: RECURSION (JAVA)
//NOTE: Class name was changed for organization.

import java.util.*;

public class Day9 {

    static int factorial(int n) {
        if (n == 1) {
            return 1;
        }
        else {
            return (factorial(n-1)*n);
        }
    }

    public static void main(String[] args) {
        Scanner in = new Scanner(System.in);
        int n = in.nextInt();
        int result = factorial(n);
        System.out.println(result);
    }
}
